rust/src/lib.rs

```diff
@@ -206,15 +264,25 @@ pub extern "C" fn zcashlc_init_accounts_table(
                 unsafe { *capacity_ret.as_mut().unwrap() = v.capacity() };
                 let p = v.as_mut_ptr();
                 std::mem::forget(v);
```

Comment:
I don't think this is a correct pattern. The [doc for `std::mem::forget`](https://doc.rust-lang.org/std/mem/fn.forget.html) says
> Any resources the value manages, such as heap memory or a file handle, will linger forever in an unreachable state. However, it does not guarantee that pointers to this memory will remain valid.

As far as I know, the Rust allocator is free to reuse the memory that was associated with `v` as soon as it goes out of scope, i.e. at the the end of this block. `std::mem::forget` *only* prevents the destructor from being run.

Reply:
See https://github.com/zcash/zcash/blob/f84a57e71bded0774bcb63e8c205c6db6ce442e1/src/rust/src/zip339_ffi.rs#L39-L72 for an example of how to robustly return a string to C/C++ and let it manage the memory.

Reply:
As @nuttycom pointed out privately, this isn't a simple string so we probably need `Box::into_raw`.